controllers/manifest_test.go

```diff
@@ -61,5 +77,299 @@ var _ = Describe("Update Manifest CR", Ordered, func() {
+			return validateManifestSpecResource(manifest.Spec.Resource, &moduleTemplate.Spec.Data)
+		}
+		Eventually(expectManifest(hasValidSpecResource), Timeout, Interval).Should(Succeed())
 	})
 })
+
+var _ = Describe("Manifest.Spec is reset after manual update", Ordered, func() {
+	const updateRepositoryURL = "registry.docker.io/kyma-project/component-descriptors"
+
+	kyma := NewTestKyma("kyma")
+
+	module := v1beta2.Module{
+		ControllerName: "manifest",
+		Name:           NewUniqModuleName(),
+		Channel:        v1beta2.DefaultChannel,
+	}
+	kyma.Spec.Modules = append(kyma.Spec.Modules, module)
+	RegisterDefaultLifecycleForKyma(kyma)
+
+	It("update Manifest", func() {
+		// await for the manifest to be created
+		Eventually(GetManifestSpecRemote, Timeout, Interval).
+			WithArguments(ctx, controlPlaneClient, kyma, module).
+			Should(Equal(false))
+
+		manifest, err := GetManifest(ctx, controlPlaneClient, kyma, module)
+		Expect(err).ToNot(HaveOccurred())
+
+		manifestImageSpec := extractInstallImageSpec(manifest.Spec.Install)
+		manifestImageSpec.Repo = updateRepositoryURL
+
+		// is there a simpler way to update manifest.Spec.Install?
+		updatedBytes, err := json.Marshal(manifestImageSpec)
+		Expect(err).ToNot(HaveOccurred())
+		manifest.Spec.Install.Source.Raw = updatedBytes
+
+		err = controlPlaneClient.Update(ctx, manifest)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("validate Manifest", func() {
+		moduleTemplate, err := GetModuleTemplate(ctx, controlPlaneClient, module.Name, "default")
+		Expect(err).NotTo(HaveOccurred())
+
+		expectManifest := expectManifestFor(kyma)
+
+		By("checking Spec.Install")
+		hasValidSpecInstall := func(manifest *v1beta2.Manifest) error {
+			return validateManifestSpecInstall(manifest.Spec.Install, moduleTemplate)
+		}
+		Eventually(expectManifest(hasValidSpecInstall), Timeout, Interval).Should(Succeed())
+
+		By("checking Spec.Resource")
+		hasValidSpecResource := func(manifest *v1beta2.Manifest) error {
+			return validateManifestSpecResource(manifest.Spec.Resource, &moduleTemplate.Spec.Data)
+		}
+		Eventually(expectManifest(hasValidSpecResource), Timeout, Interval).Should(Succeed())
+	})
+})
+
+func findRawManifestResource(reslist []compdesc.Resource) *compdesc.Resource {
+	for _, r := range reslist {
+		if r.Name == "raw-manifest" {
+			return &r
+		}
+	}
+
+	return nil
+}
+
+//nolint:goerr113,funlen
+func validateManifestSpecInstall(manifestInstall v1beta2.InstallInfo, moduleTemplate *v1beta2.ModuleTemplate) error {
+	var (
+		manifestImageSpec        *v1beta2.ImageSpec
+		moduleTemplateDescriptor *v1beta2.Descriptor
+		err                      error
+	)
+
+	manifestImageSpec = extractInstallImageSpec(manifestInstall)
+	moduleTemplateDescriptor, err = moduleTemplate.GetDescriptor()
+	if err != nil {
+		return err
+	}
+
+	//compares the actual manifest spec.install.source.name with the corresponding values from the ModuleTemplate
+	compareManifestSourceName := func() error {
```

Comment:
I suggest having smaller functions to validate each of the spec.install.source components for better readability